packages/labs/analyzer/src/lib/lit-element/events.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @fileoverview
+ *
+ * Utilities for working with events
+ */
+
+import ts from 'typescript';
+import {Event} from '../model.js';
+
+import {LitClassDeclaration} from './lit-element.js';
+
+export const getEvents = (node: LitClassDeclaration) => {
+  const events = new Map<string, Event>();
+  const jsDocTags = ts.getJSDocTags(node);
+  if (jsDocTags !== undefined) {
+    for (const tag of jsDocTags) {
+      if (tag.tagName.text === 'fires') {
+        const {comment} = tag;
+        if (comment === undefined) {
+          continue;
+        } else if (typeof comment === 'string') {
+          const result = parseFiresTagComment(comment);
+          if (result === undefined) {
+            // TODO(justinfagnani): report syntax error?
+            continue;
+          }
+          const {name, type, description} = result;
+          // TODO(justinfagnani): how do we dereference the event type name
+          // into a TypeScript type? TypeScript will automatically do this for
+          // jsdoc tags it understands, but unfortunately @fires is not one of
+          // them.
```

Comment:
Can you (temporarily) create & insert an identifier with the type name into the AST where it would be in scope (presumably in the module), and then do a `checker.getTypeAtLocation(identifier);`?

Reply:
I'm sure something like that could work, though you might have to kick the checker to re-check the module. I hope there are some direct ways of evaluating a type expression within a scope. The built-in tags do this somehow.